dfget/core/regist/register.go

```diff
@@ -68,6 +69,10 @@ func (s *supernodeRegister) Register(peerPort int) (*RegisterResult, *errortypes
 	nodes, nLen := s.cfg.Nodes, len(s.cfg.Nodes)
 	req := s.constructRegisterRequest(peerPort)
 	for i = 0; i < nLen; i++ {
+		if s.lastRegisteredNode == nodes[i] {
+			logrus.Warnf("the last registered node is the same(%s)", nodes[i])
+			continue
+		}
```

Comment:
What's the reason for this? How and when to retry  a supernode?

Reply:
> In addition, if dfget failed to download from supernode xxx, it will migrate to other supenodes in the node list of dfget. It's unreasonable to register to the supernode that just failed, and it will bring a few questions:

> It doesn't make much sense to try again for the same supernode.
The cid of dfgetTask will not change. So two dfgetTask with same cid will be registered on the same supernode, which will cause some data confusion.